src/backend/distributed/commands/create_distributed_table.c

```diff
@@ -1166,6 +1168,22 @@ EnsureRelationCanBeDistributed(Oid relationId, Var *distributionColumn,
 }
 
 
+/*
+ * ErrorIfTemporaryTable errors out if the given table is a temporary table.
+ */
+static void
+ErrorIfTemporaryTable(Oid relationId)
+{
+	Relation relation = relation_open(relationId, AccessShareLock);
+	if (relation->rd_rel->relpersistence == RELPERSISTENCE_TEMP)
```

Comment:
Minor: we can use `get_rel_persistence `

Reply:
Thanks, it is definitely better to use that as it abstracts the details.